0_RootFS/Rootfs/bundled/testsuite/Makefile

```diff
@@ -1,45 +1,65 @@
-# Default target is `buildall`, but we've also got `cleanall` and `runall`
-buildall:
-runall:
-cleanall:
+# Default target is `build`, but we've also got `clean` and `run`
+build:
+run:
+clean:
 install:
 
-LANGUAGES=c cxx fortran go rust
-PROJECTS=hello_world dyn_link openmp
+LANGUAGES :=
+
+define enable_language
+ifneq ($(shell which $(2)),)
+LANGUAGES += $(1)
+build-$(1):
+run-$(1):
+clean-$(1):
+install-$(1):
+endif
+endef
+
+$(eval $(call enable_language,c,cc))
+$(eval $(call enable_language,cxx,c++))
+$(eval $(call enable_language,fortran,gfortran))
+$(eval $(call enable_language,go,go))
+$(eval $(call enable_language,rust,rustc))
 
 define unesc
-$(subst -,/,$(subst install-,,$(subst build-,,$(subst clean-,,$(subst run-,,$1)))))
+$(subst -,/,$1)
 endef
 
-# Helper function that adds $(2) as a dependency to rule $(1), but only if $(2) actually exists.
+# Helper function that adds $(2) as a dependency to rule $(1)
 define add_dep
-ifneq ($(wildcard $(call unesc,$(2))/.),)
 $(1): $(2)
-endif
 endef
 
-# For each language, and each project within that language, add it:
-$(foreach p,$(PROJECTS),$(foreach l,$(LANGUAGES),$(eval $(call add_dep,buildall,build-$(l)-$(p)))))
-$(foreach p,$(PROJECTS),$(foreach l,$(LANGUAGES),$(eval $(call add_dep,cleanall,clean-$(l)-$(p)))))
-$(foreach p,$(PROJECTS),$(foreach l,$(LANGUAGES),$(eval $(call add_dep,runall,build-$(l)-$(p)))))
-$(foreach p,$(PROJECTS),$(foreach l,$(LANGUAGES),$(eval $(call add_dep,install,install-$(l)-$(p)))))
+# For each language, and each project within that language, add them to the overall `build-$(lang)` targets
+$(foreach l,$(LANGUAGES),$(foreach p,$(wildcard $(l)/*),$(eval $(call add_dep,build-$(l),build-project-$(l)-$(notdir $(p))))))
+$(foreach l,$(LANGUAGES),$(foreach p,$(wildcard $(l)/*),$(eval $(call add_dep,clean-$(l),clean-project-$(l)-$(notdir $(p))))))
+$(foreach l,$(LANGUAGES),$(foreach p,$(wildcard $(l)/*),$(eval $(call add_dep,install-$(l),install-project-$(l)-$(notdir $(p))))))
+
+# Add all languages to the overall `build` and `clean` targets
+$(foreach l,$(LANGUAGES),$(eval $(call add_dep,build,build-$(l))))
+$(foreach l,$(LANGUAGES),$(eval $(call add_dep,clean,clean-$(l))))
+$(foreach l,$(LANGUAGES),$(eval $(call add_dep,install,install-$(l))))
 
-build-%:
-	@$(MAKE) -sC $(call unesc,$@)
+build-project-%:
+	@$(MAKE) -sC $(call unesc,$*)
```

Comment:
Why this change? 🤔

Reply:
Because I want to have language-wide targets like `build-c`, and I don't want it to match `build-%`.